### DISCUSSION TITLE
How to get an inline hologram of the MNIST dataset?

### DISCUSSION THREAD

**Discussion by Sud11111** [rafael-fuente/Diffraction-Simulations--Angular-Spectrum-Method#19] [Q&A] [answered]
Hey there, thanks for creating such a helpful library.

I am working on numerical simulation of a paper including use of deep learning neural networks for reconstruction of holograms. I am trying to prepare the training dataset by producing inline-holograms of the images in the MNIST handwritten digits dataset.

This is what I am trying to achieve:-
![image](https://user-images.githubusercontent.com/88573156/175757170-da043418-7607-45c5-930a-0db73b60322d.png)

"The object aperture (transverse
dimension) was assumed to be 25λ × 25λ, and the recording
plane was assumed to be 30λ away from the object plane. The
lateral extent of the recorded holograms was limited to 42λ ×
42λ. All the simulations for this work were done at λ = 600 nm.
The results, however, are invariant to the wavelength given that
all the dimensions are scaled proportional to λ."

Credits-https://doi.org/10.1021/acsphotonics.1c01365?urlappend=%3Fref%3DPDF&jav=VoR&rel=cite-as

Here is my attempt at the problem:-
```
import diffractsim
diffractsim.set_backend("CPU") 
from diffractsim import MonochromaticField,ApertureFromImage, mm, cm, nm

F = MonochromaticField(
    wavelength=600*nm, extent_x=25*600 * nm, extent_y=25*600 * nm, Nx=28, Ny=28, intensity = 0.1
)

F.add(ApertureFromImage("0.png", image_size=(25*600 * nm, 25 *600* nm), simulation = F))


F.propagate(z=30*600*nm)
rgb = F.get_colors()

F.plot_colors(rgb, xlim=[-42*600*nm, 42*600*nm], ylim=[-42*600*nm, 42*600*nm])
```
Please suggest what I am doing wrong.

**Comment by rafael-fuente** [comment, marked as answer]
Hi Sud11111, 

Your holograms are amplitude holograms, and they are recorded from [-1,1] amplitude range. You need to specify this ApertureFromImage object because the default is [0,1]:

```python

import diffractsim
diffractsim.set_backend("CPU") 
from diffractsim import MonochromaticField,ApertureFromImage, mm, cm, nm

λ = 600 * nm

F = MonochromaticField(
    wavelength=600*nm, extent_x=70*λ, extent_y=70*λ, Nx=1024, Ny=1024, intensity = 0.01
)

F.add(ApertureFromImage("9.png", image_size=(25*λ, 25 *λ), simulation = F, amplitude_mask_extent = [-1,1]))


F.propagate(z=30*λ)
rgb = F.get_colors()

F.plot_colors(rgb)
```

This example would reconstruct the number 9 hologram:

![9](https://user-images.githubusercontent.com/37953893/177354021-df89af14-ad5b-45fd-8142-fb71bd73f493.png)

Note that the holograms don't have good quality, at least by using the resolution of the image you attached.
